### PR TITLE
stdalgos (2/10): team-level API, and rst update for release 4.2

### PR DESCRIPTION
**Pull request by fnrizzi**
Add team-level API to and convert to rst the following algorithms:
```
Kokkos_Equal.hpp
Kokkos_Find.hpp
Kokkos_FindIf.hpp
Kokkos_FindIfNot.hpp
Kokkos_FindEnd.hpp
Kokkos_FindFirstOf.hpp
```

NOTE: this follows the new structure for documenting the API, but does not use sphinx because the lack of properly aligning things make things quite unreadable.

Open individual html with Firefox (must be firefox or it won't display correctly) [Archive.zip](https://github.com/kokkos/kokkos-core-wiki/files/12737016/Archive.zip)

Example of rendered `find_if`:

image

image

image

### CODE REVIEW

**Review by cedricchevalier19 — APPROVED** [review]
I am not sure about how useful are the examples with explicit OpenMP.

File: docs/source/API/algorithms/std-algorithms/all/StdEqual.rst [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+   view_type b("b", 15);
+   // fill a,b somehow
+
+   // create functor
+   IsEqualFunctor<int,int> p();
+
+   bool isEqual = KE::equal(exespace, KE::begin(a), KE::end(a),
+                            KE::begin(b), KE::end(b) p);
+
+   // assuming OpenMP is enabled, then you can also explicitly call
```
Comment (dalg24) [inline docs/source/API/algorithms/std-algorithms/all/StdEqual.rst:250]: Use `Kokkos::DefaultHostExecutionSpace` instead

File: docs/source/API/algorithms/std-algorithms/all/StdEqual.rst [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+   bool isEqual = KE::equal(Kokkos::OpenMP(), KE::begin(a), KE::end(a),
```
Comment (crtrott) [inline docs/source/API/algorithms/std-algorithms/all/StdEqual.rst:251]:
```suggestion
   // To run explicitly on the host (assuming a and b are accessible on Host)
   bool isEqual = KE::equal(Kokkos::DefaultHostExecutionSpace(), KE::begin(a), KE::end(a),
```

File: docs/source/API/algorithms/std-algorithms/all/StdEqual.rst [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+Returns true if two ranges or two rank-1 ``View`` s are equal.
+
+Interface
+---------
+
+.. warning:: This is currently inside the ``Kokkos::Experimental`` namespace.
+
+
+Overload set accepting execution space
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ajpowelsnl) [inline docs/source/API/algorithms/std-algorithms/all/StdEqual.rst:19]: Is it safe to assume that `execution space` can be Host or Device?

**Review by crtrott — APPROVED** [review]

**Pull request merged**